File: AVc — копия/app/src/main/java/com/hfad/AVc/ui/database/ContactDao.java
package com.hfad.AVc.ui.database;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface ContactDao {
    @Query("SELECT * FROM contact")
    List<Contact> getAll();

    @Query("SELECT * FROM contact WHERE (favorite != 0)")
    List<Contact> getComingCongratulations();

    @Query("SELECT * FROM contact WHERE favorite != 0")
    List<Contact> getAllCongratulations();

    @Query("SELECT * FROM contact WHERE id = :id")
    Contact getById(String id);

    @Insert
    void insert(Contact contact);

    @Update
    void update(Contact contact);

    @Delete
    void delete(Contact contact);



}

File: AVc/app/src/main/java/com/hfad/avc/data/database/ContactDao.kt
package com.hfad.avc.data.database

import androidx.room.*
import com.hfad.avc.data.model.Contact

@Dao
interface ContactDao {
   @Query("SELECT * FROM contact")
   fun all(): List<Contact>

   @Query("SELECT * FROM contact WHERE (favorite != 0)  & (date_congratulations <= :secondsDeadLine)")
   fun getComingCongratulations(secondsDeadLine: Long): List<Contact>

   @Query("SELECT * FROM contact WHERE (worked != 0)")
   fun getAllCongratulations(): List<Contact>

   @Query("SELECT * FROM contact WHERE (favorite != 0)")
   fun getFavoriteContact(): List<Contact>

   @Query("SELECT * FROM contact WHERE id = :id")
   fun getById(id: String): Contact

   @Insert
   fun insert(contact: Contact)

   @Update
   fun update(contact: Contact)

   @Delete
   fun delete(contact: Contact)
}